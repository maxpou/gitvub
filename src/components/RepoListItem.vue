<template>
  <div class="card">
    <div class="card-body">
      <h4 class="card-title">{{ repository.name }}</h4>
      <p class="card-text">{{ repository.description }}</p>

      <router-link :to="{ name: 'RepoDetail', params: { user: repository.owner.login, repository: repository.name }}"
        class="btn btn-outline-info btn-block btn-sm"
      >details</router-link>
    </div>
  </div>
</template>

<script>
export default {
  name: 'RepoListItem',
  props: ['repository']
}
</script>
