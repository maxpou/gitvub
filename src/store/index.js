import Vue from 'vue'
import Vuex from 'vuex'
import repositories from './modules/repositories'
import users from './modules/users'
import navigator from './modules/navigator'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

const store = new Vuex.Store({
  modules: {
    repositories,
    users,
    navigator
  },
  strict: debug
})

/* istanbul ignore if */
if (module.hot) {
  module.hot.accept([
    './modules/repositories',
    './modules/users'
  ], () => {
    const repositories = require('./modules/repositories').default
    store.hotUpdate({
      modules: {
        repositories,
        users,
        navigator
      }
    })
  })
}

export default store
