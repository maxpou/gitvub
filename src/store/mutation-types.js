export const RECEIVE_REPOSITORIES = 'RECEIVE_REPOSITORIES'
export const RECEIVE_LANGUAGES = 'RECEIVE_LANGUAGES'
export const FETCH_USER_REPOS = 'FETCH_USER_REPOS'
export const FETCH_REPO_DETAIL = 'FETCH_REPO_DETAIL'
export const RECEIVE_REPOSITORY_LANGUAGE = 'RECEIVE_REPOSITORY_LANGUAGE'
export const RECEIVE_REPOSITORY_README = 'RECEIVE_REPOSITORY_README'
