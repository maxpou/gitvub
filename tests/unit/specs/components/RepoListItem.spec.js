import { mount } from 'vue-test-utils'
import RepoListItem from '@/components/RepoListItem.vue'
import store from '@/store'

describe('RepoListItem.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(RepoListItem, {
      store,
      propsData: {
        repository: {
          name: 'test-repo',
          description: 'i\'m a test repo',
          owner: {
            login: 'bob'
          }
        }
      }
    })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
