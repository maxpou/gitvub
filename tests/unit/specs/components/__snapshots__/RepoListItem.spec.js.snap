// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoListItem.vue test initial rendering 1`] = `"<div class=\\"card\\"><div class=\\"card-body\\"><h4 class=\\"card-title\\">test-repo</h4> <p class=\\"card-text\\">i'm a test repo</p> <router-link to=\\"[object Object]\\" class=\\"btn btn-outline-info btn-block btn-sm\\">details</router-link></div></div>"`;
